### PR TITLE
Add gather_nd and its tests

### PR DESCRIPTION
**Pull request by yeliang2258**
1.增加gather_nd的转换和对应的单测。
相关issue：https://github.com/PaddlePaddle/Knover/issues/106
2.修复gaussion_random的一个bug，新增了相关单测。
3.修复range和lookup_table的squeeze op相关bug

### CODE REVIEW

**Comment by fadelma** [comment]
Hello, I got this error when trying to convert the NSP model of PLATO-2 32L
```
2022-01-11 04:33:16 [INFO]      ONNX model generated is valid.
Traceback (most recent call last):
  File "/mnt/data3/shared_envs/paddle_env/lib/python3.9/site-packages/paddle2onnx-0.9.0-py3.9.egg/paddle2onnx/utils.py", line 43, in check_model
  File "/mnt/data3/shared_envs/paddle_env/lib/python3.9/site-packages/onnx/checker.py", line 101, in check_model
    protobuf_string = model.SerializeToString()
ValueError: Message onnx.ModelProto exceeds maximum protobuf size of 2GB: 6530750548

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/mnt/data3/shared_envs/paddle_env/bin/paddle2onnx", line 33, in <module>
    sys.exit(load_entry_point('paddle2onnx==0.9.0', 'console_scripts', 'paddle2onnx')())
  File "/mnt/data3/shared_envs/paddle_env/lib/python3.9/site-packages/paddle2onnx-0.9.0-py3.9.egg/paddle2onnx/command.py", line 187, in main
  File "/mnt/data3/shared_envs/paddle_env/lib/python3.9/site-packages/paddle2onnx-0.9.0-py3.9.egg/paddle2onnx/command.py", line 151, in program2onnx
  File "/mnt/data3/shared_envs/paddle_env/lib/python3.9/site-packages/paddle2onnx-0.9.0-py3.9.egg/paddle2onnx/convert.py", line 83, in program2onnx
  File "/mnt/data3/shared_envs/paddle_env/lib/python3.9/site-packages/paddle2onnx-0.9.0-py3.9.egg/paddle2onnx/convert.py", line 37, in export_onnx
  File "/mnt/data3/shared_envs/paddle_env/lib/python3.9/site-packages/paddle2onnx-0.9.0-py3.9.egg/paddle2onnx/graph/onnx_graph.py", line 234, in export_proto
  File "/mnt/data3/shared_envs/paddle_env/lib/python3.9/site-packages/paddle2onnx-0.9.0-py3.9.egg/paddle2onnx/utils.py", line 45, in check_model
Exception: ONNX model is not valid.
```

File: tests/test_auto_scan_gather_nd.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+        def generator_data():
+            input_data = randtool("int", 0, 10, input2_shape)
+            return input_data
+
+        config = {
+            "op_names": ["gather_nd"],
+            "test_data_shapes": [input_shape, generator_data],
+            "test_data_types": [[dtype], [dtype2]],
+            "opset_version": [15],
```
Comment (jiangjiajun) [inline tests/test_auto_scan_gather_nd.py]: opset应该也要包括11？

**Review by jiangjiajun — CHANGES_REQUESTED** [review]

File: tests/test_auto_scan_gather_nd.py (hunk 1 above)
Comment (yeliang2258) [reply tests/test_auto_scan_gather_nd.py]: done

**Comment by yeliang2258** [comment]
> Hello, I got this error when trying to convert the NSP model of PLATO-2 32L
> 
> ```
> 2022-01-11 04:33:16 [INFO]      ONNX model generated is valid.
> Traceback (most recent call last):
>   File "/mnt/data3/shared_envs/paddle_env/lib/python3.9/site-packages/paddle2onnx-0.9.0-py3.9.egg/paddle2onnx/utils.py", line 43, in check_model
>   File "/mnt/data3/shared_envs/paddle_env/lib/python3.9/site-packages/onnx/checker.py", line 101, in check_model
>     protobuf_string = model.SerializeToString()
> ValueError: Message onnx.ModelProto exceeds maximum protobuf size of 2GB: 6530750548
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/mnt/data3/shared_envs/paddle_env/bin/paddle2onnx", line 33, in <module>
>     sys.exit(load_entry_point('paddle2onnx==0.9.0', 'console_scripts', 'paddle2onnx')())
>   File "/mnt/data3/shared_envs/paddle_env/lib/python3.9/site-packages/paddle2onnx-0.9.0-py3.9.egg/paddle2onnx/command.py", line 187, in main
>   File "/mnt/data3/shared_envs/paddle_env/lib/python3.9/site-packages/paddle2onnx-0.9.0-py3.9.egg/paddle2onnx/command.py", line 151, in program2onnx
>   File "/mnt/data3/shared_envs/paddle_env/lib/python3.9/site-packages/paddle2onnx-0.9.0-py3.9.egg/paddle2onnx/convert.py", line 83, in program2onnx
>   File "/mnt/data3/shared_envs/paddle_env/lib/python3.9/site-packages/paddle2onnx-0.9.0-py3.9.egg/paddle2onnx/convert.py", line 37, in export_onnx
>   File "/mnt/data3/shared_envs/paddle_env/lib/python3.9/site-packages/paddle2onnx-0.9.0-py3.9.egg/paddle2onnx/graph/onnx_graph.py", line 234, in export_proto
>   File "/mnt/data3/shared_envs/paddle_env/lib/python3.9/site-packages/paddle2onnx-0.9.0-py3.9.egg/paddle2onnx/utils.py", line 45, in check_model
> Exception: ONNX model is not valid.
> ```

Hello, According to the official documentation of onnx, you can turn off the model check function during conversion, and then use the following command to check the model after the model is saved.
Model conversion command:
> ```
>paddle2onnx --model_dir paddle_model  --save_file onnx_file --opset_version 11 --enable_onnx_checker False
> ```

Model check command:
> ```
>import onnx
>onnx.checker.check_model('path/to/the/model.onnx')
> ```
Related document：https://github.com/onnx/onnx/blob/main/docs/PythonAPIOverview.md#checking-a-large-onnx-model-2gb
image

**Review by jiangjiajun — APPROVED** [review]

**Pull request merged**